logviewer/src/main/java/org/togetherjava/logwatcher/views/logs/StreamedView.java

```diff
@@ -0,0 +1,153 @@
+import com.vaadin.flow.router.PageTitle;
+import com.vaadin.flow.router.Route;
+import com.vaadin.flow.server.SessionDestroyEvent;
+import com.vaadin.flow.server.VaadinService;
+import org.togetherjava.logwatcher.accesscontrol.AllowedRoles;
+import org.togetherjava.logwatcher.accesscontrol.Role;
+import org.togetherjava.logwatcher.constants.LogEventsConstants;
+import org.togetherjava.logwatcher.entities.LogEvent;
+import org.togetherjava.logwatcher.logs.LogRepository;
+import org.togetherjava.logwatcher.views.MainLayout;
+import org.togetherjava.logwatcher.watcher.StreamWatcher;
+import org.vaadin.crudui.crud.impl.GridCrud;
+
+import javax.annotation.security.PermitAll;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * The Logs View in the Browser
+ */
+
+@PageTitle("Streamed")
+@Route(value = "streamed", layout = MainLayout.class)
+@AllowedRoles(roles = {Role.USER})
+@PermitAll
+public class StreamedView extends HorizontalLayout {
+
+    private final GridCrud<LogEvent> grid = new GridCrud<>(LogEvent.class);
+    private final UUID uuid = UUID.randomUUID();
+
+    public StreamedView(LogRepository logs) {
+        addClassName("hello-world-view");
+
+        add(new Button("Change Columns", this::onChangeColumns), this.grid);
+
+        this.grid.setOperations(logs::findAll, null, null, null);
+        this.grid.setAddOperationVisible(false);
+        this.grid.setDeleteOperationVisible(false);
+        this.grid.setUpdateOperationVisible(false);
+
+
+        this.grid.getGrid()
+            .setColumns(LogEventsConstants.FIELD_INSTANT, LogEventsConstants.FIELD_LOGGER_NAME,
+                    LogEventsConstants.FIELD_MESSAGE);
+        setInstantFormatter();
+        this.grid.getGrid().getColumns().forEach(c -> c.setAutoWidth(true));
+        this.grid.getGrid().getColumns().forEach(c -> c.setResizable(true));
+        this.grid.getGrid().recalculateColumnWidths();
+        this.grid.getGrid().setColumnReorderingAllowed(true);
+
+        VaadinService.getCurrent().addSessionDestroyListener(this::onDestroy);
+        final UI ui = UI.getCurrent();
+        StreamWatcher.addSubscription(this.uuid, () -> ui.access(this.grid::refreshGrid));
+    }
+
+
+    private void onDestroy(SessionDestroyEvent event) {
+        removeHook();
+    }
+
+    @Override
+    protected void onDetach(DetachEvent detachEvent) {
+        removeHook();
+        super.onDetach(detachEvent);
+    }
+
+    private void removeHook() {
+        StreamWatcher.removeSubscription(this.uuid);
+    }
+
+    private void setInstantFormatter() {
+        final Grid<LogEvent> innerGrid = this.grid.getGrid();
+        final Optional<Grid.Column<LogEvent>> column =
+                Optional.ofNullable(innerGrid.getColumnByKey(LogEventsConstants.FIELD_INSTANT));
+        if (column.isEmpty())
+            return;
```

Comment:
Missing curly braces. I am surprised that this passes our checks and style settings.
```suggestion
        if (column.isEmpty()) {
            return;
        }
```

Reply:
It seems our current style does not force it and unfortunately there is no easy way to enable it. I would still propose to change it though.

Reply:
Changed it